egs/material/s5/local/chain/decode_test.sh

```diff
@@ -97,10 +113,66 @@ done
 frames_per_chunk=$(echo $chunk_width | cut -d, -f1)
 rm $dir/.error 2>/dev/null || true
 
-for data in $test_sets; do
-  (
-    nspk=$(wc -l <data/${data}_hires/spk2utt)
-    steps/nnet3/decode.sh \
+if [ $stage -le 18 ]; then
+  for data in $test_sets; do
+    (
+      nspk=$(wc -l <data/${data}_hires/spk2utt)
+      steps/nnet3/decode.sh \
+        --acwt 1.0 --post-decode-acwt 10.0 \
+        --extra-left-context $chunk_left_context \
+        --extra-right-context $chunk_right_context \
+        --extra-left-context-initial 0 \
+        --extra-right-context-final 0 \
+        --frames-per-chunk $frames_per_chunk \
+        --nj $nspk --cmd "$decode_cmd"  --num-threads 4 \
+        --online-ivector-dir exp/nnet3${nnet3_affix}/ivectors_${data}_hires_stage1 \
+        $tree_dir/graph data/${data}_hires ${dir}/decode_${data}_stage1 || exit 1
+    ) || touch $dir/.error &
+    wait
+    # resolve ctm overlaping regions, and compute wer
+    cp $datadev/reftext data/${data}_hires
+    local/postprocess_test.sh --test-sets $test_sets \
+                              --tree-dir $tree_dir \
+                              --dir $dir \
+                              --stage1 true
+  done
+fi
+wait
+[ -f $dir/.error ] && echo "$0: there was a problem while decoding" && exit 1
+
+if [ $stage -le 19 ]; then
```

Comment:
@danpovey I extracted ivectors based on the uniform segments and vad weights in this stage and decode in the next stage. But WER was not reduced. Am I doing things right?